File: 4SumII.cs
public class Solution {
    public int FourSumCount(int[] nums1, int[] nums2, int[] nums3, int[] nums4) {
        Dictionary<int, int> mapa = new Dictionary<int, int>();
        
        foreach(int i in nums1) {
            foreach(int j in nums2) {
                if(mapa.ContainsKey(i+j)) {
                    mapa[i+j]++;
                } else {
                    mapa[i+j] = 1;
                }
            }
        }
        
        int counter = 0;
        foreach(int i in nums3) {
            foreach(int j in nums4) {
                if(mapa.TryGetValue(-(i+j), out int c))
                counter += c;
            }
        }
        return counter;
    }
}

File: 4SumII.cpp
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        std::unordered_map<int, int> mapa;
        
        for(int j=0; j<nums2.size(); j++) {
            for(int k=0; k<nums3.size(); k++) {
                mapa[nums2[j] + nums3[k]]++;
            }
        }
        
        int rez{};
        for(int j=0; j<nums1.size(); j++) {
            for(int m=0; m<nums4.size(); m++) {
                rez += mapa[- (nums1[j] + nums4[m])];
            }
        }
        return rez;
    }
};
